src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -289,6 +289,38 @@
  * }
  * </pre>
  * <p>
+ * To configure the check so that it limit IllegalType to validate field types only:
+ * </p>
+ * <pre>
+ * &lt;module name="IllegalType"&gt;
+ *   &lt;property name=&quot;illegalClassNames&quot; value=&quot;java.util.Optional&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name=&quot;query&quot; value=&quot;//METHOD_DEF//*&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * import java.util.Optional;
+ *
+ * class Optional {
+ *
+ *   private Optional&lt;String&gt; field1; // violation, usage of type 'Optional' is not allowed
+ *   static int var = 4; // OK
+ *   protected notOptional&lt;String&gt; field2; // OK
+ *   public Optional&lt;String&gt; field3; // violation, usage of type 'Optional' is not allowed
+ *   Optional&lt;String&gt; field4; // violation, usage of type 'Optional' is not allowed
+ *
+ *   void foo() {
+ *     Optional&lt;String&gt; i; // OK
+ *   }
+ *
+ *   public &lt;T extends java.util.Optional&gt; void Method(T t){
+ *     Optional&lt;String&gt; i; // OK
+ *   }
+ * }
```

Comment:
Code examples should be compilable.

Reply:
i think its bcoz of `<` written as `&lt;`

Reply:
i think its bcoz of `<` written as `&lt;`
and you can try code from here https://github.com/checkstyle/checkstyle/pull/9233#issuecomment-772274273

Reply:
Please share CLI output of this code compiling.

Reply:
```
import java.util.Optional;

class Optional {

  private Optional<String> field1; // violation, usage of type 'Optional' is not allowed
  static int var = 4; // OK
  protected notOptional<String> field2; // OK
  public Optional<String> field3; // violation, usage of type 'Optional' is not allowed
  Optional<String> field4; // violation, usage of type 'Optional' is not allowed

  void foo() {
    Optional<String> i; // OK
  }

  public <T extends java.util.Optional> void Method(T t){
    Optional<String> i; // OK
  }
}
```

```
Starting audit...
[ERROR] /home/aryaniiit002/Desktop/checkstyle/.IgnoreFor_Checkstyle/SomeClass.java:5:11: Usage of type 'Optional' is not allowed. [IllegalType]
[ERROR] /home/aryaniiit002/Desktop/checkstyle/.IgnoreFor_Checkstyle/SomeClass.java:8:10: Usage of type 'Optional' is not allowed. [IllegalType]
[ERROR] /home/aryaniiit002/Desktop/checkstyle/.IgnoreFor_Checkstyle/SomeClass.java:9:3: Usage of type 'Optional' is not allowed. [IllegalType]
Audit done.
Checkstyle ends with 3 errors.

```

Reply:
You have only shown that you can run Checkstyle on this code. [ Please compile the code example using `javac`](https://beginnersbook.com/2013/05/first-java-program/).   This code will not compile.

Reply:
adding `public`  like this
`public class Optional {`
is this okay?

```
$javac Test.java
Test.java:1: error: cannot find symbol
public class Test extends TreeSet { // OK
                          ^
  symbol: class TreeSet
Test.java:12: error: cannot find symbol
  public void fullName(TreeSet a) {} // OK
                       ^
  symbol:   class TreeSet
  location: class Test
Test.java:4: error: cannot find symbol
    LinkedHashMap<Integer, String> lhmap = new LinkedHashMap<Integer, String>(); // OK
    ^
  symbol:   class LinkedHashMap
  location: class Test
Test.java:4: error: cannot find symbol
    LinkedHashMap<Integer, String> lhmap = new LinkedHashMap<Integer, String>(); // OK
                                               ^
  symbol:   class LinkedHashMap
  location: class Test
4 errors
```

Reply:
adding `public` like this
`public class Optional {
`
is this okay?

```
$javac Optional.java
Optional.java:1: error: Optional is already defined in this compilation unit
import java.util.Optional;
^
Optional.java:5: error: type Optional does not take parameters
  private Optional<String> field1; // violation, usage of type 'Optional' is not allowed
                  ^
Optional.java:7: error: cannot find symbol
  protected notOptional<String> field2; // OK
            ^
  symbol:   class notOptional
  location: class Optional
Optional.java:8: error: type Optional does not take parameters
  public Optional<String> field3; // violation, usage of type 'Optional' is not allowed
                 ^
Optional.java:9: error: type Optional does not take parameters
  Optional<String> field4; // violation, usage of type 'Optional' is not allowed
          ^
Optional.java:12: error: type Optional does not take parameters
    Optional<String> i; // OK
            ^
Optional.java:16: error: type Optional does not take parameters
    Optional<String> i; // OK
            ^
7 errors
```

Reply:
I would suggest changing the class name to `Test` and replace the `notOptional<String> field2` to `HashSet<String> notOptionalField2`. Import the HashSet to make it compilable.

Reply:
i was also thinking of changing the class name but `notOptional`  and `field` with number (at the end) seems like in a flow.

Reply:
@AkMo3 please check and tell if u want any changes